### ISSUE TITLE
KotlinGenerator produces invalid syntax for enums with special characters

### ISSUE THREAD

**Issue by BorisOsipov** [jOOQ/jOOQ#10837]
### Expected behavior
Generated code
```
enum class GenderEnum(@get:JvmName("literal") val literal: String) : EnumType {
    gender_Male("Gender.Male"),
    gender_Female("Gender.Female")
```
### Actual behavior
Generated code
```
enum class GenderEnum(@get:JvmName("literal") val literal: String) : EnumType {
    `Gender.Male`("Gender.Male"),
    `Gender.Female`("Gender.Female"),
```
### Steps to reproduce the problem
DDL sample
```
CREATE TYPE foo.gender_enum AS ENUM (
    'Gender.Male',
    'Gender.Female',
    'Gender.NotSpecified'
    );
```
Use `org.jooq.codegen.KotlinGenerator` generator.

### Details
Dots and some other symbols aren't allowed as identifier name as described in [Kotlin language specification](https://kotlinlang.org/spec/syntax-and-grammar.html#identifiers)
I would sugest don't use backtiks function names at all. Also Kotlin code style required it

> In tests (and only in tests), it's acceptable to use method names with spaces enclosed in backticks.

See [code style ](https://kotlinlang.org/docs/reference/coding-conventions.html#function-names)

### Versions
- jOOQ: 3.14.1
- Java: 11
- Database (include vendor): PostgreSQL 10.7
- OS: Windows
- JDBC Driver (include name if inofficial driver): org.postgresql:postgresql:42.2.9

**Comment by lukaseder** [comment]
Thanks a lot for your report and for the pointer to the language specs. That's an interesting nuance to this feature set, quite different from Scala's approach.

Will look into this, soon.

**Comment by lukaseder** [comment]
The specification isn't super clear about what's possible and what isn't:
https://kotlinlang.org/spec/syntax-and-grammar.html#grammar-rule-QuotedSymbol

According to the grammar, any symbol should be possible as a "QuotedSymbol" except:

```
<any character excluding CR, LF and '`'>
```

But then, there's a vague remark about things not being possible *in some cases*:

> Note: for example, the following characters are not allowed in identifiers used as declaration names on the JVM platform even when escaped due to JVM restrictions: (, ), {, }, [, ], .

This isn't a kotlin restriction, but a kotlin-on-jvm restriction.

For the sake of this bugfix, I suspect we can just replace any of the above listed characters by an underscore.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.2 (#10845)

**Comment by lukaseder** [comment]
It appears this hasn't been fixed entirely correctly. In the meantime, I've found a better specification:
https://kotlinlang.org/docs/reference/grammar.html#Identifier

See also:

- https://github.com/jOOQ/jOOQ/issues/17099
